### DISCUSSION TITLE
guid for Visual Studio command prompt/PowerShell Profile

### DISCUSSION THREAD

**Discussion by ShinjiShioda** [microsoft/terminal#12782] [Q&A] [answered]
The guid used for the Visual studio command prompt/PowerShell profile is different for each PC.

PC1
---
    {
                "guid": "{cdb555b5-6aa8-559d-bd38-f98c8179d1f4}",
                "hidden": true,
                "name": "Developer Command Prompt for VS 2019",
                "source": "Windows.Terminal.VisualStudio"
    },

PC2
---
     {
                "guid": "{5a9f3981-6fa2-599f-a4ca-8f99e6159450}",
                "hidden": false,
                "name": "Developer Command Prompt for VS 2019",
                "source": "Windows.Terminal.VisualStudio"
    },

If we have different guids for the same commandLine program, we have to change the key assignment in Action and the description in json fragment for each PC.
For example, in the following case

in "actions"
---
        {"command":  {
                "action": "newTab",
                "profile": "{46ca431a-3a87-5fb3-83cd-11ececc031d2}"
            }, "keys": "ctrl+shift+k" },

in settings.json json fragment
---
    "profiles": [
      {
        // windows powershell 5.1
        "updates": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
        "backgroundImage": "WindowsPowerShell.png",
        "backgroundImageStretchMode": "uniform",
        "suppressApplicationTitle":true,
        "tabTitle":"wPs"
      },
----
In the above cases, by using guid, the installation status of the program on each PC is not affected.
A copy of the entire settings.json may not match the installation status of the command line program on the individual PC.

Is it possible to make the GUID in the Visual studio profile the same between PCs as in WSL?

**Comment by zadjii-msft** [comment]
I'm gonna try and tag in @heaths on this one. I forget how we're coming up with the GUIDs for the VS profiles. I think the WSL ones are based solely on the distro's name, so those are constant across machines.

**Reply by DHowett** [reply]
I suspect we are using the individual instance ID from VSSetup...

**Comment by heaths** [comment, marked as answer]
Yes, the GUID is based on the instance ID which is automatically generated when installed.

https://github.com/microsoft/terminal/blob/0bc66abd3b9777200178ca2f4760eaee7ab72812/src/cascadia/TerminalSettingsModel/VsDevCmdGenerator.cpp#L19

The instance ID must be unique across different Visual Studio installations to avoid collisions: the same VS edition can be installed multiple times, which comes in handy for previewing different channels without disrupting team standards. Perhaps seeding the GUID with the instance ID wasn't ideal, but I'm not sure there is an ideal seed. Can't use the display name because it's not unique (for the reason mentioned above) and using the installation path could be misleading because you might have different editions or channels installed to the same path on two different machines and the profile for one may not work for the profile of another.

**Reply by ShinjiShioda** [reply]
Thanks for the reply.
I understand that it is difficult to identify VS installations by version, edition or channel.
